addons/udes_stock/tests/test_picking.py

```diff
@@ -376,3 +375,30 @@ def test08_get_suggested_location_by_product_all_lots(self):
         # and the location with matching product but different lot.
         self.assertEqual(len(suggested_locations), 2)
         self.assertEqual(self.test_location_01 + self.test_location_02, suggested_locations)
+
+    def test09_considers_partially_available_move_lines_when_suggesting_location(self):
+        Location = self.env['stock.location']
+
+        drop_policy = "by_height_speed"
+        products_info = [{"product": self.apple, "qty": 10}]
+        self.product_category_slow = self.create_category(name='Slow')
+        self.product_category_ground = self.create_category(name='Ground')
+        self.apple.u_height_category_id = self.product_category_ground
+        self.apple.u_speed_category_id = self.product_category_slow
+
+        # We need another empty location
+        self.test_location_05 = Location.create({
+            "name": "Test location 05",
+            "barcode": "LTEST05",
+            "location_id": self.stock_location.id,
+        })
+
+        picking1 = self.create_and_assign_putaway_picking(products_info, drop_policy)
+        picking2 = self.create_and_assign_putaway_picking(products_info, drop_policy)
+        picking1.apply_drop_location_policy()
+        picking1.move_line_ids.write({'state': 'partially_available'})
```

Comment:
It would be good to get it in partially_available without changing the state manually. Have we tried to increase the move quantity by one unit? something like `picking1.move_lines[0].product_uom_qty += 1`. Otherwise I would add a comment that we are changing the state manually just to keep the test simpler.

Reply:
Changed, please review.